tools/data-api/internal/endpoints/v1/common_types.go

```diff
@@ -0,0 +1,74 @@
+package v1
+
+import (
+	"fmt"
+	"github.com/hashicorp/pandora/tools/data-api/internal/repositories"
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/hashicorp/pandora/tools/data-api/models"
+)
+
+func commonTypes(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	opts, ok := ctx.Value("options").(Options)
+	if !ok {
+		internalServerError(w, fmt.Errorf("missing options"))
+		return
+	}
+	payload := models.CommonTypesDetails{
+		Constants: map[string]models.ConstantDetails{},
+		Models:    map[string]models.ModelDetails{},
+	}
+	if !opts.UsesCommonTypes {
+		render.JSON(w, r, payload)
+		return
+	}
+
+	commonTypes, ok := ctx.Value("commonTypes").(repositories.CommonTypesDetails)
+	if !ok {
+		internalServerError(w, fmt.Errorf("missing commonTypes"))
+		return
+	}
```

Comment:
This information isn't threaded through on the context, but if it were we'd be mapping this anyway (per above) to discard all of the data (lines 24-17) - instead we'll need to retrieve every Service and iterate over each API Version and each Resource to map out all of the Constants/Models.

Whilst there's going to be conflicts for Resource Manager, Graph should have uniquely named models throughout the stack (hence the toggle in `opts.UsesCommonTypes` to enable this only for MS Graph)